spec/request/api/v1/sessions/create_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'rails_helper'
+
+describe 'POST api/v1/users/sign_in', type: :request do
+  let(:password) { 'password' }
+  let(:token) do
+    {
+      '70crCAAYmNP1xLkKKM09zA' =>
+      {
+        'token' => '$2a$10$mSeRnpVMaaegCpn3AhORGe5wajFhgMoBjGIrMwq4Qq2mP6f/OHu1y',
+        'expiry' => 153_574_356_4
+      }
+    }
+  end
+  let(:user) { create(:user, password: password, tokens: token) }
+
+  context 'with correct params' do
+    before do
+      params = {
+        user:
+          {
+            email: user.email,
+            password: password
+          }
+      }
+      post new_user_session_path, params: params, as: :json
```

Comment:
The post should not be part of the `before` block, it should be the `subject` of the spec.

The `before` is usually used to prepare the test (similar to the `lets` but without actually assigning anything to a variable).

In this case I think I would do something like:
```
subject { post new_user_session_path, params: params, as: :json }

context 'with correct params' do
let(:params) do
  {
    user: {
      email: user.email,
      password: password
    }
  }
end
```

Reply:
This site about how to write specs can be a good guide for you! 💪 
https://www.betterspecs.org/

Reply:
A common structure is:
```
describe '{ACTION}' do
  subject {  {url_to_action}, params: params, as: :json } 
  
  context '{a context for the tests}' do
     let(:params) { {some_assignment_for_the_params_matching_the_context} }
     
     it 'one expectation' do
       expect(subject).to ...
     end
     
     it 'other expectation' do
       expect(subject).to ...
     end
  end
  
  context '{another context for the tests}' do
     let(:params) { {some_assignment_for_the_params_matching_the_context} }
     
     it 'one expectation' do
       expect(subject).to ...
     end
     
     it 'other expectation' do
       expect(subject).to ...
     end
  end
end
```

Reply:
It is possible that instead of `expect(subject)` you have to use `expect(subject.response)`

Another option is to do that in the `subject` definition, like:
```
subject do
  {url_to_action}, params: params, as: :json
  response
end
```